javascript.md

```diff
@@ -1,11 +1,56 @@
-# Javascript
+# JavaScript
 
-JavaScript at P'unk Avenue follows a consistent style. What follows is a summary of that style. These are practices that are always followed in our npm modules and should ideally be followed in project-level and other non-Apostrophe code as well.
+Use P'unk Ave's [ESLint configuration](https://www.npmjs.com/package/eslint-config-punkave) (based on JavaScript Standard Style) to lint all JavaScript with your editor or during a build process. Install [Atom's ESLint plugin](https://github.com/AtomLinter/linter-eslint).
 
-Let's start with the basics.
+## Table of Contents
 
-## ESLint
-Use P'unk Ave's [ESLint configuration](https://www.npmjs.com/package/eslint-config-punkave) (based on JavaScript Standard Style) to lint all JavaScript with your editor or during a build process. Install [Atom's ESLint plugin](https://github.com/AtomLinter/linter-eslint).
+1. [Front-end](#front-end)
+
+## Front-end
+<a name="visual-state"></a><a name="1.1"></a>
+[1.1](#visual-state) Add and remove classes to manage visual state
+
+> Why? Provides a separation of concerns.
+
+```javascript
+// Good
+$foo.removeClass('is-active');
+// Bad
+$foo.hide();
+```
+
+<a name="data-attributes"></a><a name="1.2"></a>
+[1.2](#data-attributes) Use data attributes as event handlers
+
+> Why? Provides a separation of concerns.
+
+```javascript
+// Good
+$('[data-foo]').removeClass('is-active');
+// Bad
+$('foo').removeClass('is-active');
```

Comment:
Stylistic thing. I'd say we use `$('.foo')` as a more realistic example. A class might also look more distinct from data attributes.